dwave/optimization/include/dwave-optimization/nodes/indexing.hpp

```diff
@@ -254,6 +254,23 @@ class BasicIndexingNode : public ArrayNode {
     const bool contiguous_;
 };
 
+class ConcatenateNode : public ArrayOutputMixin<ArrayNode> {
```

Comment:
I'd put this in the new [manipulation.hpp](https://github.com/dwavesystems/dwave-optimization/blob/main/dwave/optimization/include/dwave-optimization/nodes/manipulation.hpp).

Reply:
Oh right, I forgot to move it after rebasing. Will fix!